frontend/src/components/icon.tsx

```diff
@@ -66,6 +76,7 @@ export const Icon = React.forwardRef<SVGSVGElement, React.PropsWithChildren<Icon
             <button
                 onClick={onClick}
                 disabled={busy}
+                data-test-id={dti}
                 css={Object.assign({
                     border: 'none',
                     padding: 0,
```

Comment:
you'll also need to set the `data-test-id` on the icon if the button isn't rendered, otherwise it'll be lost